### PR TITLE
Plantnet sync

### PR DESCRIPTION
**Pull request by kstefanini**

### CODE REVIEW

File: src/Command/CelSyncGetJobsCommand.php [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+        $email = (string) $input->getOption('email');
+
+        $this->plantnetPaginator->start($startDate, $email);
+
+        do {
+            $occurrences = $this->plantnetPaginator->getContent();
+            if (!$occurrences) {
+                break;
+            }
+            $occurrences = $occurrences->getData();
```
Comment (kstefanini) [inline src/Command/CelSyncGetJobsCommand.php:96]: simp

File: src/Command/CelSyncGetJobsCommand.php [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+        }
+    }
+
+    private function addJob(string $type, int $id)
+    {
+        if (!in_array($type, ['update', 'create', 'delete'])) {
+            throw new \Exception('Job has be deleted, created or updated, nothing else');
+        }
+
+        $changelog = $this->changeLogRepository->findOneBy(
```
Comment (kstefanini) [inline src/Command/CelSyncGetJobsCommand.php]: line

File: src/Command/CelSyncProcessJobsCommand.php [hunk 3]
```diff
@@ -0,0 +1,283 @@
...
+use App\Entity\PnTbPair;
+use App\Model\AnnuaireUser;
+use App\Service\AnnuaireService;
+use App\Service\IdentiplanteService;
+use App\Service\OccurrenceBuilderService;
+use App\Service\PhotoBuilderService;
+use App\Service\PlantnetService;
+use App\Service\TaxoRepoService;
+use Doctrine\ORM\EntityManagerInterface;
+use phpDocumentor\Reflection\Types\This;
```
Comment (kstefanini) [inline src/Command/CelSyncProcessJobsCommand.php]: wtf

File: src/Command/CelSyncGetPnTbPairsIdsCommand.php [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+            $this->em->flush();
+        }
+        $bar->finish();
+
+        $event = $stopwatch->stop('pn-tb-pairs-ids');
+        $this->io->success(
+            sprintf('PN TB pairs successfully synced! (Stats: %s new pairs | %s known occurrences | %s unknown)',
+                $savedPairs, $knownOccurrences, $unknownOccurrences)
+        );
+        $this->io->comment(sprintf('Elapsed time: %.2f ms / Consumed memory: %.2f MB', $event->getDuration(), $event->getMemory() / (1024 ** 2)));
```
Comment (kstefanini) [inline src/Command/CelSyncGetPnTbPairsIdsCommand.php:102]: return ?

File: src/Command/CelSyncGetJobsCommand.php [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+
+        $event = $stopwatch->stop('pn-sync-get-jobs');
+        if ($output->isVerbose()) {
+            $this->io->success(sprintf(
+                'Success! Got %d new jobs, %d already know, out of %d total processed occurrences!',
+                $this->newJobsCount, $this->existingJobsCount, count($occurrences)
+            ));
+
+            $this->io->comment(sprintf('Elapsed time: %.2f ms / Consumed memory: %.2f MB', $event->getDuration(), $event->getMemory() / (1024 ** 2)));
+        }
```
Comment (kstefanini) [inline src/Command/CelSyncGetJobsCommand.php:131]: return ?

File: src/Repository/PlantnetOccurrenceRepository.php [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+use Doctrine\Bundle\DoctrineBundle\Repository\ServiceEntityRepository;
+use Symfony\Bridge\Doctrine\RegistryInterface;
+
+/**
+ * @method PlantnetOccurrence|null find($id, $lockMode = null, $lockVersion = null)
+ * @method PlantnetOccurrence|null findOneBy(array $criteria, array $orderBy = null)
+ * @method PlantnetOccurrence[]    findAll()
+ * @method PlantnetOccurrence[]    findBy(array $criteria, array $orderBy = null, $limit = null, $offset = null)
+ */
+class PlantnetOccurrenceRepository extends ServiceEntityRepository
```
Comment (kstefanini) [inline src/Repository/PlantnetOccurrenceRepository.php]: wtf

File: src/Service/AnnuaireService.php [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+
+    public function isKnownUser(string $email): bool
+    {
+        return (bool)$this->findUserInfo($email);
+    }
+
+    private function fixDumbAnnuaireDataStructure(string $data): string
+    {
+        // switch from "{"killian@tela-botanica.org":{"id":"1312","prenom":null,"nom":null,"pseudo":null,"pseudoUtilise":false,"intitule":"killian-stefanini","groupes":[],"nomWiki":"Killianstefanini"}}"
+        // to
```
Comment (kstefanini) [inline src/Service/AnnuaireService.php]: to what?

**Review by idiralliche — APPROVED** [review]

**Comment by kstefanini** [comment]
First edit config env
Now run migration for new PnTbPair table, run command to get pairs, then run getJobs command for newly modified occurrences (2-3 days old) and processJobs, for load testing

**Pull request merged**